notebooks/21_reinforcement_learning/index.md

```diff
@@ -0,0 +1,522 @@
+
+<img src="resources/modelbased.png" align="left" width="315px"/>
+
+| | &ensp;&ensp;&ensp;&ensp;&ensp;Episode1  &ensp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;Episode2  &emsp;&emsp;&emsp;|
+|-| ----|
+|| <table>   <tbody>    <td>B</td>  <td>east</td> <td>C</td> <td>-1</td>      <td>&emsp;&emsp;&emsp;&ensp;&ensp;B</td>     <td> east</td>    <td> C</td>  <td> -1</td>   </tr>  <tr>  <td>C</td>  <td>east</td> <td>D</td><td>-1</td> <td>&emsp;&emsp;&emsp;&ensp;&ensp;C</td>     <td> east</td>    <td> D</td>  <td> -1</td>     </tr>  <tr>  <td>D</td>  <td>exit</td> <td>X</td><td>+10</td>  <td>&emsp;&emsp;&emsp;&ensp;&ensp;D</td>     <td> exit</td>    <td> X</td>  <td> +10</td>  </tr>  <tr>    </tbody>  </table>            
+
+| | &ensp;&ensp;&ensp;&ensp;&ensp;Episode3  &ensp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;Episode4  &emsp;&emsp;&emsp;|
+|-| ----|
+|| <table>   <tbody>    <td>E</td>  <td>north</td> <td>C</td> <td>-1</td>      <td>&emsp;&emsp;&emsp;&ensp;&ensp;E</td>  <td>north</td> <td>C</td> <td>-1</td>   </tr>  <tr>  <td>C</td>  <td>east</td> <td>D</td><td>-1</td> <td>&emsp;&emsp;&emsp;&ensp;&ensp;C</td>     <td> east</td>    <td> A</td>  <td> -1</td>     </tr>  <tr>  <td>D</td>  <td>exit</td> <td>X</td><td>+10</td>  <td>&emsp;&emsp;&emsp;&ensp;&ensp;A</td>     <td> exit</td>    <td> X</td>  <td> +10</td>  </tr>  <tr>    </tbody>  </table>     
+
+<br clear="left"/>
+<br>
+
+After sampling, in order to estimate $\hat{T}$ for each $(s,a,s')$, we have to count the number of all movements that start from $s$ and have the action $a$ and go to state $s'$. After that we have to divise this count to the number of movements that start from $s$ and have the action $a$. For example:
+* $T(B, east, C) = \frac{2}{2} = 1$
+* $T(C, east, D) = \frac{3}{4} = 0.75$
+<br>
+In addition, we can estimate $\hat{R}(s,a,s')$ for each movement in all episodes. We have to get the average of the reward for all tuples of $(s,a,s')$. For example:
+<br>
+* $$R(B, east, C) = \frac{-1-1}{2} = -1$$
+* $$R(D, exit, x) = \frac{10+10+10}{3} = 10$$
+
+**Question:** What is the problem of above model-based algorithm?<br> If the number of states are large, the number of required samples to train the model will be $O((|S|)^2.|A|)$. In other words, we need many episodes to be able to train the model properly. 
+
+<div id='TypesofReinforcementLearningAccordingtoLearningPolicy'/>
+
+## Types of Reinforcement Learning According to Learning Policy
+Ther are two types of RL according to Learning policy: **Passive Reinforcement Learning**, **Active Reinforcement Learning**. In below, we explain both methods.

```

Comment:
Avoid typos, please... also, ameliorate comprehension by changing the structure: "According to Learning policy, there are two types of RL: Passive Reinforcement Learning and Active Reinforcement Learning."

Reply:
Do you mean that we should only change the structure of that specific sentence?